### PR TITLE
Fix S3 snapshots to work with newer versions of K3s/RKE2 pt. 2

### PR DESCRIPTION
**Pull request by Oats87**
Follow up of: https://github.com/rancher/rancher/pull/37398

https://github.com/rancher/rancher/issues/34417

### CODE REVIEW

**Review by snasovich — APPROVED** [review]
LGTM, but I'll be honest I'm not fully understanding the logic here so definitely need @thedadams approval before merging.

**Review by thedadams — APPROVED** [review]

**Pull request merged**
